docs/website/scripts/generate_index.py

```diff
@@ -0,0 +1,138 @@
+from pathlib import Path
+from string import Template
+
+import tomllib
+from typing import NamedTuple, Sequence
+
+from generate_header import generate_navigation
+
+
+class Paper(NamedTuple):
+    title: str
+    abstract: str
+    pdf: str
+    arxiv: str
+    venue: str
+    year: int
+    authors: str
+
+
+class Framework(NamedTuple):
+    name: str
+    repository: str
+    icon: str
+    summary: str
+    papers: Sequence[Paper]
+    documentation: str = None
+
+
+def load_framework_definitions(definition_file: Path = Path("official_frameworks.toml")) -> list[Framework]:
+    with definition_file.open("rb") as fh:
+        frameworks = tomllib.load(fh)["frameworks"]
+
+    return [
+        Framework(
+            **{attr: val for attr, val in fw.items() if attr != "papers"},
+            papers=tuple(Paper(**paper) for paper in fw.get("papers", [])),
+        )
+        for fw in frameworks
+    ]
+
+
+def load_navigation() -> str:
+    return generate_navigation()
+
+
+def load_footer() -> str:
+    with open("templates/footer.html", "r") as f:
```

Comment:
replace string with defined constant.
why are some templates in files and others hard-coded?